torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -418,19 +433,100 @@ void ONNXSetDynamicInputShape(
   }
 }
 
+bool SequenceTypeOutput(Node* node) {
+  if (node->kind() == ::c10::onnx::SplitToSequence ||
+      node->kind() == ::c10::onnx::SequenceInsert ||
+      node->kind() == ::c10::onnx::SequenceEmpty ||
+      node->kind() == ::c10::onnx::SequenceConstruct)
+    return true;
+  return false;
+}
+
+void ONNXInferOutputShapeType(
```

Comment:
together with the below comment, this can be renamed to something like `ONNXUpdateTypeFromTensor`. this may be applied to input as well in the future.